packaging/build_rpm.py

```diff
@@ -0,0 +1,375 @@
+########
+# Copyright (c) 2017 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+#    * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    * See the License for the specific language governing permissions and
+#    * limitations under the License.
+
+import argparse
+import logging
+import sys
+import shutil
+import urllib2
+from datetime import datetime
+from os import chdir, listdir
+from os.path import (
+        abspath,
+        basename,
+        dirname,
+        getmtime,
+        join as path_join,
+        split as path_split,
+        )
+from subprocess import check_call, check_output, CalledProcessError
+
+
+MOCK = '/usr/bin/mock'
+LOCAL_REPO_PATH = '~/mock_repo'
+DEPENDENCIES_FILE = '{spec_file}.dependencies'
+RESULT_DIR = '/var/lib/mock/epel-7-x86_64/result'
+
+SCRIPT_DIR, SCRIPT_NAME = path_split(abspath(__file__))
+SSH_CONFIG_FILE = path_join(SCRIPT_DIR, 'ssh_config.tmp')
+
+RPMS_DIR = SCRIPT_DIR
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(SCRIPT_NAME)
+
+
+def main(args):
+    parser = argparse.ArgumentParser(
+            description="Build RPM using mock",
```

Comment:
I guess it can be vagrant as well, right?
Maybe we should even make it explicit?

Reply:
it used to be an explicit `--local` for non-vagrant. Hmm. Personally i think i'd prefer the vagrant part to be a separate script

Reply:
@tehasdf so that I have to run a different command to you on my dev machine vs your dev VM?
My aim is to have a system test runner script (or plugin, whatever) (when the system tests get merged into `cloudify-manager`) which can do the RPM builds as necessary and then run the tests. I don't want users of that script (or even the script itself) to need to worry about whether vagrant is a requirement or not.
Also everything that happens in `main()` before the mock/vagrant condition makes more sense to run on the host if there is a host.